src/simulation/simulation_models.jl

```diff
@@ -0,0 +1,141 @@
+"""
+Stores the OperationProblem definitions to be used in the simulation. When creating the
+SimulationModels, the order in which the models are created determines the order on which
+the simulation is executed.
+"""
+mutable struct SimulationModels
+    decision_models::Vector{<:DecisionModel}
+    emulation_model::Union{Nothing, EmulationModel}
+
+    function SimulationModels(
+        decision_models::Vector,
+        emulation_model::Union{Nothing, EmulationModel} = nothing,
+    )
+        all_names = [get_name(x) for x in decision_models]
+        emulation_model !== nothing && push!(all_names, get_name(emulation_model))
+        model_count =
+            emulation_model === nothing ? length(decision_models) :
+            length(decision_models) + 1
+        if length(Set(all_names)) != model_count
+            error("All model names must be unique: $all_names")
+        end
+
+        return new(decision_models, emulation_model)
+    end
+end
+
+function SimulationModels(
+    decision_models::DecisionModel,
+    emulation_model::Union{Nothing, EmulationModel} = nothing,
+)
+    return SimulationModels([decision_models], emulation_model)
+end
+
+function SimulationModels(;
+    decision_models,
+    emulation_model::Union{Nothing, EmulationModel} = nothing,
+)
+    return SimulationModels(decision_models, emulation_model)
+end
+
+function get_simulation_model(models::SimulationModels, name::Symbol)
```

Comment:
can we make a method like this for `get_simulation_model(sim::Simulation, name::Symbol)`?

Reply:
... and export this function

Reply:
exported in ed40a47